hazelcast/src/main/java/com/hazelcast/query/impl/LocalDateHelper.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.query.impl;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Calendar;
+import java.util.Date;
+
+public final class LocalDateHelper {
+
+    private static final String[] DATE_FORMATS = new String[] { "yyyy-MM-dd", "yyyy/MM/dd", "yyyy-MM-dd hh:mm:ss",

```

Comment:
Not sure, is it possible to use here the predefined formats from `DateTimeFormatter` instead, like `ISO_DATE`, and possibly locale specific ones as a fallback, like `DateTimeFormatter.ofLocalizedDateTime()`?

Reply:
I am not sure how we can use `DateTimeFormatter.ofLocalizedDateTime()` since it supports only few formats [here](https://github.com/frohoff/jdk8u-dev-jdk/blob/master/src/share/classes/java/time/format/FormatStyle.java). however if you know any list of string formats for datetime we can simply add here.

Reply:
@taburet Could you please help @omidp there?